Scripts/Core/VRView.cs

```diff
@@ -189,6 +184,70 @@ public void OnEnable()
             m_CameraRig.position = headCenteredOrigin;
             m_CameraRig.rotation = Quaternion.identity;
 
+            if (s_ExistingSceneMainCamera)
+            {
+                var cameraGameObject = m_Camera.gameObject;
+                var potentialImageEffects = s_ExistingSceneMainCamera.GetComponents<MonoBehaviour>();
+                var enabledPotentialImageEffects = potentialImageEffects.Where(x => x != null && x.enabled);
+                var targetMethodNames = new [] {"OnRenderImage", "OnPreRender", "OnPostRender"};
+                var bindingFlags = BindingFlags.Public | BindingFlags.NonPublic | BindingFlags.Instance | BindingFlags.Static;
+                foreach (var potentialImageEffect in enabledPotentialImageEffects)
+                {
+                    var componentInstanceType = potentialImageEffect.GetType();
+                    var targetMethodFound = false;
+                    for (int i = 0; i < targetMethodNames.Length; ++i)
+                    {
+                        // Each of the three checks is performed to catch the various image effect variants I've tested against
+                        // Each check catches a different case that was encountered during testing
+                        // Check isntanced type for target methods
+                        targetMethodFound = componentInstanceType.GetMethod(targetMethodNames[i], bindingFlags) != null;
+
+                        // Check base type for target methods
+                        if (!targetMethodFound)
+                        {
+                            var componentBaseType = componentInstanceType.BaseType;
+                            if (componentBaseType != null)
+                                targetMethodFound = componentBaseType.GetMethod(targetMethodNames[i], bindingFlags) != null;
+                        }
+
+                        // Check nested types for target methods
+                        if (!targetMethodFound)
+                        {
+                            var nestedTypes = componentInstanceType.GetNestedTypes();
+                            foreach (var nestedType in nestedTypes)
+                            {
+                                if (nestedType == null)
+                                    continue;
+
+                                targetMethodFound = nestedType.GetMethod(targetMethodNames[i], bindingFlags) != null;
+                                if (targetMethodFound)
+                                    break;
+                            }
+                        }
```

Comment:
Will the nested types actually get their methods called? What image effect is this necessary for?

Reply:
I can't find the exact effect that this handled for.  I've since deleted the many effects that I tested against.  Though, after removing this nested type check, the new roster of effects I'm testing with all work as expected (except for Amplify Occlusion, which still crashes).  I'll commit the removal of this block.